### PR TITLE
Typing: Trainer

### PR DESCRIPTION
**Pull request by asi1024**
Depends on #269.

Reopen #288.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Let me try running a test.
/test

File: pytorch_pfn_extras/training/_trainer.py [hunk 1]
```diff
@@ -24,39 +36,45 @@ def __init__(self, handler, *, evaluator, **kwargs):
...
 
-    def is_epoch_last_iter(self, idx):
+    def is_epoch_last_iter(self, idx: int) -> bool:
         return (idx + 1) == (self.manager._iters_per_epoch)
 
-    def _complete_step(self, idx, outs, *, is_deferred=False):
+    def _complete_step(
+            self,
+            idx: int,
+            outs: 'handler_module.Outputs',
```
Comment (emcastillo) [inline pytorch_pfn_extras/training/_trainer.py:74]: I looked for this definition in:

https://github.com/pfnet/pytorch-pfn-extras/blob/master/pytorch_pfn_extras/handler.py

but it wasn't there. Can you double check please?

Comment (kmaehashi) [reply pytorch_pfn_extras/training/_trainer.py:74]: Oh, I expected mypy to complain via `/test` but not...?

Comment (emcastillo) [reply pytorch_pfn_extras/training/_trainer.py:74]: yeah, I also expected it, but test came out clean

Comment (asi1024) [reply pytorch_pfn_extras/training/_trainer.py:74]: That is because `mypy: ignore-errors` was added to master branch.

**Comment by asi1024** [comment]
Fixed. PTAL.

**Comment by emcastillo** [comment]
LGTM!

**Pull request merged**